File: Leetcode100/leetcode100/src/main/java/com/jiang/leetcode100/house_robber/HouseRobber.java
package main.java.com.jiang.leetcode100.house_robber;

import java.util.Scanner;

class Solution {
    public int rob(int[] nums) {
        if(nums.length == 0) return 0;
        int[] dp = new int[nums.length];
        for(int i = 0; i < nums.length; i++) {
            if(i == 0) dp[i] = nums[i];
            else if(i == 1) dp[i] = Math.max(dp[i - 1], nums[i]);
            else dp[i] = Math.max(dp[i - 1], dp[i - 2] + nums[i]);
        }
        return dp[nums.length - 1];
    }
}

public class HouseRobber {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String str = scanner.nextLine();
        scanner.close();
        String[] nums_str = str.split(" ");
        int[] nums = new int[nums_str.length];
        for(int i = 0; i < nums_str.length; i++) nums[i] = Integer.parseInt(nums_str[i]);

        System.out.println(new Solution().rob(nums));
    }
}

File: Leetcode100/leetcode100_python/HouseRobber.py
from typing import List

class Solution:
    def rob(self, nums: List[int]) -> int:
        if not nums:
            return 0
        if len(nums) == 1:
            return nums[0]
        dp0, dp1 = nums[0], max(nums[0], nums[1])
        for i in range(2, len(nums)):
            dp0, dp1 = dp1, max(dp0 + nums[i], dp1)
        return dp1

if __name__ == "__main__":
    nums = [int(num) for num in input().split()]
    solution = Solution()
    print(solution.rob(nums))